editoast/src/infra_cache.rs

```diff
@@ -1014,71 +1135,158 @@ pub mod tests {
         }
     }
 
-    pub fn create_switch_type_point() -> SwitchType {
+    pub fn create_switch_type_point<T: AsRef<str>>(
+        id: T,
+        groups: HashMap<String, Vec<SwitchPortConnection>>,
+    ) -> SwitchType {
         SwitchType {
-            id: "point".into(),
+            id: id.as_ref().into(),
             ports: vec!["BASE".into(), "LEFT".into(), "RIGHT".into()],
-            groups: HashMap::from([
-                (
-                    "LEFT".into(),
-                    vec![create_switch_connection("BASE".into(), "LEFT".into())],
-                ),
-                (
-                    "RIGHT".into(),
-                    vec![create_switch_connection("BASE".into(), "RIGHT".into())],
-                ),
-            ]),
+            groups,
```

Comment:
I don't understand this change

Reply:
I did these changes in order to produce some wrong switch_type_point for the errors on a switch type

Reply:
What about keeping this function producing a valid switch type and modify the returned value to create errors when necessary?